lib/tools.js

```diff
@@ -68,67 +68,64 @@ function asString (str) {
 }
 
 function asJson (obj, msg, num, time) {
-  // to catch both null and undefined
-  const hasObj = obj !== undefined && obj !== null
-  const objError = hasObj && obj instanceof Error
-  msg = !msg && objError === true ? obj.message : msg || undefined
+  var objectWithMessage = Object.assign({}, obj)
+  const objError = obj instanceof Error
+  // Object.assign makes a shallow copy
```

Comment:
`stack` isn't not copied because of shallow copy - it's at the top level 
it's not copied because it's non-enumerable (which is why we have to explicitly output the stack in the logs)